### PR TITLE
ENH: Add tool for modifying repo properties

### PR DESCRIPTION
**Pull request by tangkong**
With the advent of custom repository properties, we should have some way of setting these programmatically.  This borrows a lot from the branch protection work.  

Some notes:
- This differs from the branch protection helper by not changing any property that isn't specified.  These properties already have defaults, so it's plausible that we might want to only change one or two properties with this tool
- `argparse.BooleanOptionalAction` is new and nifty to me.
- Custom properties does not have a graphql api, so I ended up using a package called `ghapi`.  The package itself is quite neat, and just requires a PAT with fine grained access to the custom properties for this to work.  As I was looking back, I realized most of our existing tooling just calls `gh api` in a subprocess, so I could have extended that.  But the `ghapi` call was so short and clean...

```bash
$ python update_repo_properties.py --help
usage: update_repo_properties.py [-h] [--repo_type [{None,Python Library,TwinCAT Library,PLC,Backup,EPICS IOC,External,Python Dev,EPICS module,Other}]]
                                 [--status_checks [{Python Standard,TwinCAT Standard,None}]] [--protect-master | --no-protect-master] [--protect-pages | --no-protect-pages]
                                 [--protect-default | --no-protect-default] [--repo-data-path REPO_DATA_PATH] [--write]
                                 [owner] [repo_name]

positional arguments:
  owner                 Organization or owner of the repository, "pcdshub" by default
  repo_name             Name of the repository

optional arguments:
  -h, --help            show this help message and exit
  --repo_type [{None,Python Library,TwinCAT Library,PLC,Backup,EPICS IOC,External,Python Dev,EPICS module,Other}]
                        Type of the repository. (Default: Other)
  --status_checks [{Python Standard,TwinCAT Standard,None}]
                        Status checks to require before merging a Pull Request. (Default: None)
  --protect-master, --no-protect-master
                        Add master protection. This should be applied as much as possible. This rule requires pull requests and disallows force pushes.
  --protect-pages, --no-protect-pages
                        Add gh-pages branch protection, allowing force pushes to the gh-pages branch specifically
  --protect-default, --no-protect-default
                        Add default (*) branch protection, disallows the creation of new branches
  --repo-data-path REPO_DATA_PATH
                        Path to repo data csv. Expects columns for: ['repo_name', 'owner', 'repo_type', 'default', 'master', 'gh_pages', 'status_checks']
  --write               Apply changes. Default is a dry run
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 apischema
 black
+ghapi
```
Comment (ZLLentz) [inline requirements.txt:3]: Is this a candidate for inclusion in pcds-envs?

File: update_repo_properties.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+@dataclass
+class RepoProperty:
+    """field names match table headers exactly"""
+    repo_name: str
+    owner: str = 'pcdshub'
+    repo_type: Optional[RepoType] = None
+    default: Optional[bool] = None
+    master: Optional[bool] = None
+    gh_pages: Optional[bool] = None
+    status_checks: Optional[StatusChecks] = None
```
Comment (ZLLentz) [inline update_repo_properties.py:60]: Do we need to update this script if we change our property names?
Do we need to update this script if another group in the same org adds their own property names?

**Review by ZLLentz — APPROVED** [review]
This looks pretty comprehensive and clean

File: update_repo_properties.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("--repo_type", type=str, nargs='?',
+                        choices=list(t.value for t in RepoType),
+                        help='Type of the repository. (Default: Other)')
+    parser.add_argument("--status_checks", type=str, nargs='?',
+                        choices=list(t.value for t in StatusChecks),
+                        help="Status checks to require before merging a Pull Request."
+                             " (Default: None)")
+    parser.add_argument("--protect-master", action=argparse.BooleanOptionalAction,
```
Comment (ZLLentz) [inline update_repo_properties.py:208]: One more thing, just a comprehension question:
Does this mean:

- no arg provided -> no change
-  `--protect-master` provided -> set to `True`
- `--no-protect-master` provided -> set to `False`

Comment (tangkong) [reply update_repo_properties.py:208]: That's correct. If the flag isn't supplied, the argument is `None`

File: update_repo_properties.py (hunk 2 above)
Comment (tangkong) [reply update_repo_properties.py:60]: We will have to update this if we change our property names, yes.

If other groups in the org add their own properties, I imagine we'll just have to agree not to change each other's props. As it is the script will just change the ones requested.

We'll probably have to change the names to something prepended with "ecs_", but that's a bridge we can cross later

File: requirements.txt (hunk 1 above)
Comment (tangkong) [reply requirements.txt:3]: I'd argue it's venv-able. I'm starting to be very reluctant to add packages to pcds conda haha. I also think I don't expect everyone to be making gh api calls frequently

**Pull request merged**